packages/lesswrong/server/collections/advisorRequests/mutations.ts

```diff
@@ -0,0 +1,150 @@
+
+import schema from "@/lib/collections/advisorRequests/newSchema";
+import { accessFilterSingle } from "@/lib/utils/schemaUtils";
+import { userCanDo, userOwns } from "@/lib/vulcan-users/permissions";
+import { runCountOfReferenceCallbacks } from "@/server/callbacks/countOfReferenceCallbacks";
+import { logFieldChanges } from "@/server/fieldChanges";
+import { getDefaultMutationFunctions } from "@/server/resolvers/defaultMutations";
+import { getCreatableGraphQLFields, getUpdatableGraphQLFields } from "@/server/vulcan-lib/apollo-server/graphqlTemplates";
+import { makeGqlCreateMutation, makeGqlUpdateMutation } from "@/server/vulcan-lib/apollo-server/helpers";
+import { checkCreatePermissionsAndReturnProps, checkUpdatePermissionsAndReturnProps, insertAndReturnCreateAfterProps, runFieldOnCreateCallbacks, runFieldOnUpdateCallbacks, updateAndReturnDocument } from "@/server/vulcan-lib/mutators";
+import { dataToModifier } from "@/server/vulcan-lib/validation";
+import gql from "graphql-tag";
+import clone from "lodash/clone";
+import cloneDeep from "lodash/cloneDeep";
+
+
+function newCheck(user: DbUser | null, document: CreateAdvisorRequestDataInput | null, context: ResolverContext) {
+  return userCanDo(user, [
+    'advisorrequest.create',
+    'advisorrequests.new',
+  ]);
+}
+
+function editCheck(user: DbUser | null, document: DbAdvisorRequest | null, context: ResolverContext) {
+  if (!user || !document) return false;
+
+  // If we have legacy action permissions defined
+  // for this collection, check if user can perform
+  // the current action based on whether they own the
+  // document or not.  Otherwise, check if user is an
+  // admin.
+  return userOwns(user, document as HasUserIdType)
+    ? userCanDo(user, [
+      'advisorrequest.update.own',
+      'advisorrequests.edit.own',
+    ])
+    : userCanDo(user, [
+      'advisorrequest.update.all',
+      'advisorrequests.edit.all',
+    ]);
+}
+
+
+const { createFunction, updateFunction } = getDefaultMutationFunctions('AdvisorRequests', {
+  createFunction: async ({ data }: CreateAdvisorRequestInput, context, skipValidation?: boolean) => {
+    const { currentUser } = context;
+
+    const callbackProps = await checkCreatePermissionsAndReturnProps('AdvisorRequests', {
+      context,
+      data,
+      schema,
+      skipValidation,
+    });
+
+    data = callbackProps.document;
+
+    data = await runFieldOnCreateCallbacks(schema, data, callbackProps);
+
+    const afterCreateProperties = await insertAndReturnCreateAfterProps(data, 'AdvisorRequests', callbackProps);
+    let documentWithId = afterCreateProperties.document;
+
+    await runCountOfReferenceCallbacks({
```

Comment:
I think `runCountOfReferenceCallbacks` should be called by a wrapper provided by `getDefaultMutationFunctions`, not manually. It's never safe to omit this because whether it's necessary depends on other collections with foreign-key relations to this one; so omitting it shouldn't be an option.